### PR TITLE
FORMS-8029 - Static Text Core Component should update value via Rules

### PR DESCRIPTION
**Pull request by TalmizAhmed**
## Description
https://jira.corp.adobe.com/browse/FORMS-8029

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/text/v1/text/clientlibs/site/js/textview.js [hunk 1]
```diff
@@ -61,6 +61,14 @@
             setFocus() {
                 this.setActive();
             }
+
+            updateValue(value) {
+                // html sets undefined value as undefined string in input value, hence this check is added
+                let actualValue = typeof value === "undefined" ? null :  value;
```
Comment (deepprakash345) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/text/v1/text/clientlibs/site/js/textview.js:67]: in what scenario will we get undefined as value from model ?
Also if the value is undefined? is putting null recommended or should we put an empty string?will it cause any difference in behaviour?

**Review by rismehta — CHANGES_REQUESTED** [review]
Please add a test case for this use-case

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/text/v1/text/clientlibs/site/js/textview.js [hunk 2]
```diff
@@ -61,6 +61,14 @@
             setFocus() {
                 this.setActive();
             }
+
+            updateValue(value) {
```
Comment (deepprakash345) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/text/v1/text/clientlibs/site/js/textview.js:65]: Can you add a cypress test to verify this behaviour

**Review by deepprakash345 — CHANGES_REQUESTED** [review]

File: ui.tests/test-module/specs/text/text.runtime.spec.js [hunk 3]
```diff
@@ -74,7 +74,7 @@
         })
     });
 
-    it('test the rules editor', () => {
+    it.only('test the rules editor', () => {
```
[the remark this answers is not in the document]
Comment (TalmizAhmed) [reply ui.tests/test-module/specs/text/text.runtime.spec.js:77]: My bad

**Review by rismehta — APPROVED** [review]

**Pull request merged**